docs/AI-for-security/knowledge-base.asciidoc

```diff
@@ -0,0 +1,139 @@
+[[ai-assistant-knowledge-base]]
+= AI Assistant Knowledge Base
+
+AI Assistant's knowledge base feature enables it to recall specific documents and other specified information, and to use it as context when responding to your queries. This page describes how to enable and add information to knowledge base.
+
+NOTE: When you upgrade from {elastic-sec} version 8.15 to a newer version, information previously stored by AI Assistant will be lost. 
+
+.Requirements
+[sidebar]
+--
+
+* To use knowledge base, you need the `Elastic AI Assistant: All` privilege. To edit global knowledge base entries (information that will affect the AI Assistant experience for other users in the {kib} space), you need the `Allow Changes to Global Entries` privilege. 
+* To use knowledge base, you must <<ml-requirements, enable machine learning>> with a minimum ML node size of 4GB.
+
+--
+
+[discrete]
+[[enable-knowledge-base]]
+== Enable knowledge base
+
+There are two ways to enable knowledge base.
+
+NOTE: You must individually enable knowledge base for each {kib} space where you want to use it.
+
+[discrete]
+=== Option 1 — enable knowledge base from an AI Assistant conversation
+
+Open a conversation with AI Assistant, select a large language model, then click **Setup Knowledge Base**. If the button doesn't appear, knowledge base is already enabled.
+
+image::images/knowledge-base-assistant-setup-button.png[An AI Assistant conversation showing the Setup Knowledge Base button]
+
+Knowledge base setup may take several minutes. It will continue in the background if you close the conversation. After setup is complete, you can access knowledge base settings from the conversation settings menu.
+
+image::images/knowledge-base-assistant-menu-dropdown.png[AI Assistant's dropdown menu with the knowledge base option highlighted]
+
+[discrete]
+=== Option 2 — enable knowledge base from the Security AI settings menu
+
+. To open Security AI settings, use the {kibana-ref}/introduction.html#kibana-navigation-search[global search field] to find "AI assistant for security".
+. Click **Setup Knowledge Base**. If the button doesn't appear, knowledge base is already enabled.
+
+image::images/knowledge-base-assistant-settings-kb-tab.png[AI Assistant's settings menu open to the knowledge base tab]
+
+[discrete]
+[[rag-for-alerts]]
+== Knowledge base for alerts
+When knowledge base is enabled, AI Assistant receives alerts from your environment as context for each of your prompts. It will receive alerts from the last 24 hours that have a status of `open` or `acknowledged`, ordered first by risk score, then by recency. Building block alerts are excluded. This enables it to answer questions about multiple alerts in your environment, rather than just the individual alerts you choose to include as context. 
```

Comment:
```suggestion
When Knowledge Base is enabled, AI Assistant receives alerts from your environment within the last 24 hours that have a status of `open` or `acknowledged`. It uses these as context for each of your prompts, answering questions about multiple alerts in your environment rather than just the individual alerts you choose to include as context. Alerts are ordered first by risk score, and then by the most recently generated. Building block alerts are excluded. 
```